File: frontend/lib/evaluationLabel.ts
export function evaluationLabel(evalCp: number | null, mateIn?: number | null): string {
  if (mateIn !== null && mateIn !== undefined) {
    return `Mat en ${Math.abs(mateIn)}`;
  }
  if (evalCp === null || evalCp === undefined) {
    return "Évaluation inconnue";
  }

  const absolute = Math.abs(evalCp);
  if (absolute < 30) {
    return "Position équilibrée";
  }

  const side = evalCp > 0 ? "blanc" : "noir";
  if (absolute < 80) {
    return `Léger avantage ${side}`;
  }
  if (absolute < 150) {
    return `Avantage clair ${side}`;
  }
  if (absolute < 300) {
    return `Gros avantage ${side}`;
  }
  return `Avantage décisif ${side}`;
}

export function technicalEvaluation(evalCp: number | null, mateIn?: number | null): string {
  if (mateIn !== null && mateIn !== undefined) {
    return `mate ${mateIn}`;
  }
  if (evalCp === null || evalCp === undefined) {
    return "n/a";
  }
  return `${evalCp} cp`;
}

File: backend/app/evaluation_label.py
from __future__ import annotations


def evaluation_label(eval_cp: int | None, mate_in: int | None = None) -> str:
    if mate_in is not None:
        return f"Mat en {abs(mate_in)}"
    if eval_cp is None:
        return "Évaluation inconnue"

    absolute = abs(eval_cp)
    if absolute < 30:
        return "Position équilibrée"

    side = "blanc" if eval_cp > 0 else "noir"
    if absolute < 80:
        return f"Léger avantage {side}"
    if absolute < 150:
        return f"Avantage clair {side}"
    if absolute < 300:
        return f"Gros avantage {side}"
    return f"Avantage décisif {side}"


def raw_eval_label(eval_cp: int | None, mate_in: int | None = None) -> str:
    if mate_in is not None:
        return f"mate {mate_in}"
    if eval_cp is None:
        return "n/a"
    return f"{eval_cp} cp"
